### PR TITLE
update the workflow ahead of upcoming deprecations

### PR DESCRIPTION
**Pull request by emmyoop**
Resolves: #15 

Replaced actions with updated versions.  Added some standardized formatting.

### CODE REVIEW

**Comment by emmyoop** [comment]
@nathaniel-may added you as a reviewer to make sure it was sane to bump the python versions to match supported versions in core.  Went ahead and dropped 3.7 since we're dropping it this quarter anyways.

File: .github/workflows/build_python.yml [hunk 1]
```diff
@@ -10,34 +20,46 @@ jobs:
...
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v1
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
       - name: Setup npm
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
```
[the remark this answers is not in the document]
Comment (nathaniel-may) [reply .github/workflows/build_python.yml:51]: it's so tree-sitter can output the js bindings

**Review by nathaniel-may — COMMENTED** [review]
Yes, updating the python versions here to match core support is a good idea. The test failures look real though.

**Comment by emmyoop** [comment]
Tests were failing because the flavor on linux we were using was deprecated and did not support python 3.11.

File: bindings/python/setup.cfg [hunk 2]
```diff
@@ -10,10 +10,10 @@ author_email = info@dbtlabs.com
...
     Programming Language :: Python :: 3 :: Only
-    Programming Language :: Python :: 3.6
-    Programming Language :: Python :: 3.7
     Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Programming Language :: Python :: 3.11
 
 [options]
 python_requires = >=3.6.1
```
Comment (nathaniel-may) [inline bindings/python/setup.cfg:19]:
```suggestion
python_requires = >=3.8.0
```

**Comment by emmyoop** [comment]
@nathaniel-may since I updated the bindings, do we need a new tag?

**Review by davidbloss — APPROVED** [review]
LGTM!

**Comment by nathaniel-may** [comment]
@emmyoop we don't need to do that till we're releasing a new version. If we added new parsing capabilities like in #14 then we would want to release with a new tag.

**Pull request merged**